### DISCUSSION TITLE
other service fails immediately because no internet is not ready

### DISCUSSION THREAD

**Discussion by jumoog** [qdm12/gluetun#1022] [Q&A] [answered]
I use `network_mode: "service:gluetun"` in my docker-compose but the other service starts immediately and has no internet and closes with exit(1). Only if I wait a minute (less maybe works as well) at startup until gluetun is completely finished everything works. The wiki says you don't need `depends_on`. But how else can I solve the problem without waiting?

**Comment by qdm12** [comment, marked as answer]
Your second container should be retryng N times for a few seconds.

This is a common problem for stacks with a database and a server containers, where the database takes 10s to boot up. In that case the server should keep on retrying for up to 1 minute.

What's your second container? If you have a shell in there, you could bind mount an entrypoint script and do a little while loop to wait for internet to be up, before exec the actual entrypoint.
